### PR TITLE
Paid/Free Confidence check experiment name update

### PR DESCRIPTION
**Pull request by ddc22**
Related to 

- https://github.com/Automattic/wp-calypso/pull/80647
- 21394-explat-experiment

#Fixes Automattic/growth-foundations#186

## Proposed Changes

* Change the experiment name to the confidence check experiment `calypso_onboarding_plans_paid_domain_on_free_plan_confidence_check`
* Fixes the domain generation where domain were not generating for some domain queries.

## Testing Instructions

* Make sure the assignments works as expected. (Should show the domain purchase allowed modal)
image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by aaronyan — APPROVED** [review]
Even though this is a draft, it LGTM!

File: packages/data-stores/src/domain-suggestions/queries.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export function useGetWordPressSubdomain( paidDomainName: string ) {
 		quantity: 1,
 		include_wordpressdotcom: true,
 		include_dotblogsubdomain: false,
-		only_wordpressdotcom: true,
+		only_wordpressdotcom: false,
```
Comment (southp) [inline packages/data-stores/src/domain-suggestions/queries.ts]: Could you help me understand why we’d need this change, and also if we could separate it into another PR if it's unrelated? 

From the code, if it needs only_wordpressdotcom to be set as false to return the correct dotcom subdomains, it needs to be fixed from the backend rather than the front-end. If that’s the case, the backend fix will then break the front-end if we ship this change. 🤔

Comment (ddc22) [reply packages/data-stores/src/domain-suggestions/queries.ts]: Right I have removed this above change for now so that we can pick this up in a separate PR.
